internal/kittehs/ptr.go

```diff
@@ -0,0 +1,9 @@
+package kittehs
+
+func ZeroIfNil[T any](t *T) (r T) {
```

Comment:
1. This function must be covered by unit tests
2. This function is used only in one place, it has no generic need, so it needs to be in/near the file in which it's used

Reply:
@daabr I see it's used in several places at `cmd/ak/cmd/mcp/tools.go`.
IMO the main issue is that the configuration parameter is defined as pointer and not as concrete value.

Reply:
removed